### ISSUE TITLE
Wdllexport-explicit-instantiation-decl warning when compiling fmt

### ISSUE THREAD

**Issue by DaanDeMeyer** [fmtlib/fmt#1181]
Getting this warning compiling with clang-cl on Windows.

```
[5/27] Building CXX object _deps\fmt-build\CMakeFiles\fmt.dir\src\posix.cc.obj
In file included from _deps\fmt-src\src\posix.cc:13:
In file included from _deps\fmt-src\include\fmt/posix.h:28:
_deps\fmt-src\include\fmt/format.h(761,1): warning: explicit instantiation declaration should not be 'dllexport' [-Wdllexport-explicit-instantiation-decl]
extern template struct basic_data<void>;
^
_deps\fmt-src\include\fmt/format.h(747,8): note: attribute is here
struct FMT_API basic_data {
       ^
_deps\fmt-src\include\fmt/core.h(160,30): note: expanded from macro 'FMT_API'
#  define FMT_API __declspec(dllexport)
                             ^
1 warning generated.
[6/27] Building CXX object _deps\doctest-build\CMakeFiles\doctest_with_main.dir\doctest\parts\doctest.cpp.obj
[7/27] Linking CXX static library _deps\doctest-build\doctest_with_main.lib
[8/27] Building CXX object _deps\fmt-build\CMakeFiles\fmt.dir\src\format.cc.obj
In file included from _deps\fmt-src\src\format.cc:8:
In file included from _deps\fmt-src\include\fmt/format-inl.h:11:
_deps\fmt-src\include\fmt/format.h(761,1): warning: explicit instantiation declaration should not be 'dllexport' [-Wdllexport-explicit-instantiation-decl]
extern template struct basic_data<void>;
^
_deps\fmt-src\include\fmt/format.h(747,8): note: attribute is here
struct FMT_API basic_data {
       ^
_deps\fmt-src\include\fmt/core.h(160,30): note: expanded from macro 'FMT_API'
#  define FMT_API __declspec(dllexport)
                             ^
1 warning generated.
[9/27] Linking CXX shared library bin\fmtd.dll
   Creating library _deps\fmt-build\fmtd.lib and object _deps\fmt-build\fmtd.exp
   Creating library _deps\fmt-build\fmtd.lib and object _deps\fmt-build\fmtd.exp
```

**Comment by vitaut** [comment]
I think this has been fixed by #1134 already.

**Comment by DaanDeMeyer** [comment]
My bad. I found that pr but didn't notice it was only merged after the last release.
